packages/vite/src/node/plugins/css.ts

```diff
@@ -548,13 +561,22 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
 
     async renderChunk(code, chunk, opts) {
       let chunkCSS = ''
+      let line = 1
+      const concatCssEndLines: Array<{ file: string; end: number }> = []
       let isPureCssChunk = true
       const ids = Object.keys(chunk.modules)
       for (const id of ids) {
         if (styles.has(id)) {
           // ?transform-only is used for ?url and shouldn't be included in normal CSS chunks
           if (!transformOnlyRE.test(id)) {
-            chunkCSS += styles.get(id)
+            const content = styles.get(id)!
+            if (chunkCSS !== '') {
+              chunkCSS += '\n' + content
```

Comment:
I think we should avoid adding the new line if the content is empty, no?
```suggestion
              if (content !== '') {
                chunkCSS += '\n' + content
              }
```

Reply:
In this case, I think `getLineCount` should return 1 instead of 0. It is mentioned before that we should split chunks of css with newline to make the mapping back clear. I believe this has to be done same to the empty css file as well.

https://stackblitz.com/edit/node-3w2ggf?file=index.js

Reply:
In the minimal repo above, `file: "foo2"` is empty, but if we just ignore the `foo2`, it would be

```
{ file: 'foo3', end: 3 },
{ file: 'foo4', end: 5 },
{ file: 'foo2', end: 5 },
{ file: 'foo1', end: 6 },
```

Even though when something errors with `foo1`, it will properly map back to `foo1`, but I think I will be better to have it like the below

```
{ file: 'foo3', end: 3 },
{ file: 'foo4', end: 5 },
{ file: 'foo2', end: 6 },
{ file: 'foo1', end: 7 },
```

Reply:
What would be your thought on this @sapphi-red ?

Reply:
If the file content is empty, no error can happen inside that and we don't need to differentiate it. So I think we can even skip `concatCssEndLines.push`.
```
{ file: 'foo3', end: 3 },
{ file: 'foo4', end: 5 },
{ file: 'foo1', end: 6 },
```

Reply:
Yup, I agree with you. I changed the logic as below.

https://stackblitz.com/edit/node-zxzubi?file=index.js